### DISCUSSION TITLE
Roam.Garden should offer similar settings for namespaces as Roam

### DISCUSSION THREAD

**Discussion by tombarys** [roam-garden/i#10] [Feature requests & Ideas]
**Describe the bug**
Roam _namespace view switching options_ are missing. I am a heavy namespace user as I use them for differentiating some special kinds of pages. Namespaces are great for internal organizing, but they are not so great to be shown outside... In this case, they can clutter the public garden and can even confuse a reader.

Roam offers a simple option to switch between levels of visibility of the namespace headers. You can switch between full view, capital letters, and completely hidden namespace/s (which I prefer for my roam-gardening ideas as it hides my internal data organization). See:

<img width="566" alt="CleanShot 2021-02-26 at 17 18 00@2x" src="https://user-images.githubusercontent.com/5729352/109325780-9eb33f80-7856-11eb-9447-c163acb5fb8d.png">

To explain how valuable namespacing can be, let's consider a Roam page with the name **Person/Elon Musk** dedicated to a known entrepreneur. You can have another page **Book/Elon Musk** about a book with the same name but with completely different information – book bibliographical data etc. Still, when you switch the namespace view off, you can see only the **Elon Musk** link in any context without everything preceding the `/`. Internally these are still two completely different pages with no special (database) connection. 

**Expected behavior**
Add Roam-garden-based setting to a Plant Garden page for switching between the visibility of all namespaces used in a public garden. Same as in Roam User Settings (picture above). Different styling not necessary.

**Comment by kalmir** [comment]
I there are two different cases when it comes to "switching between the visibility of all namespaces": links and page titles. Even Roam offers namespace prefixes to be hidden only in links, not in page titles. If Vlad implemented the hiding of prefixes in links, I think it could be a bit confusing for visitors - that two identically named links can lead to two differently named pages.

I've never felt particular need to use namespaces for differentiating kind of content – which is what most namespace users use them for, it seems. I did use them for differentiating broad topic areas. Then faced the issue how to export from this namespaced topic area without prefix for Roam.Garden. I used a bash script deleting the prefix from links and page titles which worked but was an extra step. I eventually figured it would be easiest to use a separate roam graph. Especially when it is already quite easy to link from one Roam graph to another, to the block level, via url - sometimes I want to connect notes in the new separate graph with my main one (and I don't need to embed between the two which is not yet possible).

**Reply by tombarys** [reply]
I agree it could be problematic for some use cases and user configurations. That is the reason I suggested adding (replicating) the ability to let the garden creator decide on this depending on their own namespace use. 

Another question is if this (namespace prefixes hiding) could be done via CSS only. If yes, everything is much easier as the problem for me is mostly the aestestics and possible confusion.

**Reply by JayFarei** [reply]
big user of name spaces as well for my roam graph - would love the ability to hide them from links through CSS.
